frontend/src/components/SearchBar.tsx

```diff
@@ -1,39 +1,48 @@
+import { zodResolver } from '@hookform/resolvers/zod';
 import { useForm } from 'react-hook-form';
 import { z } from 'zod';
 import { Form, FormControl, FormField, FormItem } from './ui/form';
 import { Search } from 'lucide-react';
 import { Input } from './ui/input';
-import { zodResolver } from '@hookform/resolvers/zod';
 import { Button } from './ui/button';
+import { useEffect } from 'react';
 
 const formSchema = z.object({
   searchQuery: z.string({
     required_error: 'Restaurant name is required',
   }),
 });
 
+export type SearchForm = z.infer<typeof formSchema>;
+
 type Props = {
   onSubmit: (formData: SearchForm) => void;
   placeHolder: string;
   onReset?: () => void;
+  searchQuery?: string;
 };
 
-export type SearchForm = z.infer<typeof formSchema>;
-
-const SearchBar = ({ onSubmit, placeHolder, onReset }: Props) => {
+const SearchBar = ({ onSubmit, onReset, placeHolder, searchQuery }: Props) => {
   const form = useForm<SearchForm>({
     resolver: zodResolver(formSchema),
+    defaultValues: {
+      searchQuery,
+    },
   });
 
-  function handleReset(): void {
+  useEffect(() => {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/39#issuecomment-2406845576>#a6e538</a><span> - 10/11/2024, 08:17 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The useEffect hook that updates the form when searchQuery changes may cause unnecessary re-renders and potential infinite loops.
</div>
<div id="fix">
<b>Fix:</b> Modify the useEffect hook to only update the form when the searchQuery prop changes and is different from the current form value.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -32,7 +32,9 @@
 
    useEffect(() => {
 -    form.reset({ searchQuery });
 +    if (searchQuery !== form.getValues().searchQuery) {
 +      form.reset({ searchQuery });
 +    }
    }, [form, searchQuery]);
 
    const handleReset = () => {
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged